lib/src/sv_sign.c

```diff
@@ -883,10 +886,10 @@ signed_video_set_private_key(signed_video_t *self, const char *private_key, size
     SV_THROW_IF(!self->plugin_handle, SV_EXTERNAL_ERROR);
   SV_CATCH()
   SV_DONE(status)
-
-  // Free the EVP_PKEY since it is no longer needed. It is handled by the signing plugin.
-  sv_openssl_free_key(self->sign_data->key);
-  self->sign_data->key = NULL;
+  // If ONVIF is available, call initialize Onvif.
+  if (self->onvif) {
+    status = initialize_onvif(self);
```

Comment:
This is not correct. If `status` is an error before this line and `initialize_onvif` succeeds, you will replace `status` with `SV_OK` and return it when it actually failed further up.